src/repl/interpreter.rs

```diff
@@ -70,23 +70,44 @@ impl ClarityInterpreter {
         contract_identifier: QualifiedContractIdentifier,
         cost_track: bool,
         coverage_reporter: Option<TestCoverageReport>,
-    ) -> Result<ExecutionResult, (String, Option<Diagnostic>, Option<Error>)> {
-        let mut ast = self.build_ast(contract_identifier.clone(), snippet.clone())?;
-        let (annotations, mut diagnostics) = self.collect_annotations(&ast, &snippet);
+        parser_version: u32,
+    ) -> Result<ExecutionResult, Vec<Diagnostic>> {
+        let (mut ast, mut diagnostics, success) =
+            self.build_ast(contract_identifier.clone(), snippet.clone(), parser_version);
```

Comment:
If we're not adding the "fail fast" mode in this PR (used in future blockchain environment), could you open a PR for it? Thanks!

Reply:
Sure, I opened #89 for this. It could be very simple to include in this PR, but I'd like to discuss how that should be managed, so let's do it later.